pkg/networkservice/core/adapters/client_to_server.go

```diff
@@ -14,54 +14,54 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Package adapters provides adapters to translate between networkservice.NetworkService{Server,Client}
 package adapters
 
 import (
 	"context"
 
 	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+	"google.golang.org/grpc"
 
 	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
-
-	"github.com/networkservicemesh/api/pkg/api/networkservice"
 )
 
-type clientToServer struct {
-	client networkservice.NetworkServiceClient
-}
+const clientContextKey contextKeyType = "client"
+
+type (
+	clientToServer struct {
+		client networkservice.NetworkServiceClient
+	}
+	ctxClient     struct{}
```

Comment:
Why define a type for this instead of using struct{}

Reply:
1. To implement an interface `NetworkServiceClient` on it.
1. To distinguish `type ctxClient struct{}` and `type ctxServer struct{}`. It's two different types, each with its methods.